app/containers/AudioPlayer/index.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { View } from 'react-native';
+import { AVPlaybackStatus } from 'expo-av';
+import { activateKeepAwake, deactivateKeepAwake } from 'expo-keep-awake';
+import { useSharedValue } from 'react-native-reanimated';
+import { useNavigation } from '@react-navigation/native';
+
+import { useTheme } from '../../theme';
+import styles from './styles';
+import Seek from './Seek';
+import PlaybackSpeed from './PlaybackSpeed';
+import PlayButton from './PlayButton';
+import audioPlayer from '../../lib/methods/audioPlayer';
+
+interface IAudioPlayerProps {
+	fileUri: string;
+	loading: boolean;
+	isReadyToPlay: boolean;
+	disabled?: boolean;
+	onPressCallback?: Function;
+}
+
+const AudioPlayer = ({
+	fileUri,
+	disabled = false,
+	loading = true,
+	isReadyToPlay = false,
+	onPressCallback = () => {}
+}: IAudioPlayerProps) => {
+	const [paused, setPaused] = useState(true);
+	const [rate, setRate] = useState(1);
+
+	const duration = useSharedValue(0);
+	const currentTime = useSharedValue(0);
+
+	const { colors } = useTheme();
+
+	const audioUri = useRef<string>('');
+
+	const navigation = useNavigation();
+
+	const onPlaybackStatusUpdate = (status: AVPlaybackStatus) => {
+		if (status) {
+			onPlaying(status);
+			handlePlaybackStatusUpdate(status);
+			onEnd(status);
+		}
+	};
+
+	const onPlaying = (data: AVPlaybackStatus) => {
+		if (data.isLoaded && data.isPlaying) {
+			setPaused(false);
+		} else {
+			setPaused(true);
+		}
+	};
+
+	const handlePlaybackStatusUpdate = (data: AVPlaybackStatus) => {
+		if (data.isLoaded && data.durationMillis) {
+			const durationSeconds = data.durationMillis / 1000;
+			duration.value = durationSeconds > 0 ? durationSeconds : 0;
+			const currentSecond = data.positionMillis / 1000;
+			if (currentSecond <= durationSeconds) {
+				currentTime.value = currentSecond;
+			}
+			setRate(data.rate);
+		}
+	};
+
+	const onEnd = (data: AVPlaybackStatus) => {
+		if (data.isLoaded) {
+			if (data.didJustFinish) {
```

Comment:
On `onPlaying` and `handlePlaybackStatusUpdate` you do a compound `if`, but here you chose to do two.
Unless I'm being naive, you could just

```suggestion
		if (data.isLoaded && data.didJustFinish) {
```